src/lib.rs

```diff
@@ -966,7 +1061,16 @@ impl Connection<Connected> {
         match parser::parse_control_op(reader)? {
             parser::ControlOp::Pong => Ok(()),
             parser::ControlOp::Err(e) => Err(Error::new(ErrorKind::ConnectionRefused, e)),
-            _ => Err(Error::new(ErrorKind::ConnectionRefused, "Protocol Error")),
+            other @ parser::ControlOp::Ping
+            | other @ parser::ControlOp::Msg(_)
+            | other @ parser::ControlOp::Info(_)
+            | other @ parser::ControlOp::Unknown(_) => {
```

Comment:
Even though it is unlikely these will change much for now, it's a good habit to be explicit about variants for matched enums, as it makes it easier to refactor code when you add variants. You want your code to break in relevant places when you change the variants, so that the compiler error messages point you to places where the refactor has not been accounted for yet

Reply:
What is the `other @` syntax?

Reply:
It's a way to bind the entire `ControlOp` enum to a name. It's useful more often when you want to destructure a complex type in a match but also have a name to refer to the overall complex type. Revisiting this particular case, it's probably better to just bind the `parser::parse_control_op(reader)?` call to a variable outside the match statement and refer to that variable from inside the passthrough arm. I'll clean that up now.